rfc/XXX-trusted-kubernetes-images.md

```diff
@@ -0,0 +1,148 @@
+# RFC XXX: Trusted Kubernetes Images
+
+Kubernetes control plane images should be verified by the Constellation installation.
+
+## The Problem
+
+When we bootstrap the Constellation Kubernetes cluster, `kubeadm` places a set
+of static pods for the control plane components into the filesystem. The
+manifests refer to images in a registry beyond the users' control, and the
+image content is not reproducible.
+
+This is obviously a trust issue, because the Kubernetes control plane is
+part of Constellation's TCB, but it is also a problem when Constellation is set
+up in a restricted environment where this repo is not available.
+
+## Requirements
+
+1. In a default installation, Constellation must verify Kubernetes control plane images.
+
+Out of scope:
+
+- Customization of the image repository or image content.
+  - This is orthogonal to image verification and will be subject of a separate
+    RFC.
+- Reproducibility from github.com/kubernetes/kubernetes to registry.k8s.io and
+  the associated chain of trust.
+  - It is not clear whether Kubernetes images can be reproduced at all [1].
+    Either way, the likely threat is a machine-in-the-middle attack between
+    Constellation and registry.k8s.io. A desirable addition to this proposal
+    could be verification of image signatures [2].
+- Container registry trust & CA certificates.
+  - This is also orthogonal to image verification.
+
+[1]: https://github.com/kubernetes/kubernetes/blob/master/build/README.md#reproducibility
+[2]: https://kubernetes.io/docs/tasks/administer-cluster/verify-signed-artifacts/
+
+## Solution
+
+Kubernetes control plane images are going to be pinned by hash and verified by
```

Comment:
```suggestion
Kubernetes control plane images are going to be pinned by a hash, which is verified by
```
nit: reword due to ambiguity with the above mentioned signature verification of container images